### DISCUSSION TITLE
[Deployment] foundry zksync `forge create` shows "Error: contract was not deployed" even though it is deployed

### DISCUSSION THREAD

**Discussion by coffee-the-dev** [zkSync-Community-Hub/zksync-developers#766] [Deployments] [answered]
### Team or Project

Abstract

### Environment

Testnet

### L2 block number

966244

### zkSolc Version

1.5.4

### zksync-ethers Version

N/A

### Minimal Reproducible Code or Repo Link

https://github.com/coffeexcoin/erc20-sample

### Does this work on other EVMs? (If yes, please list at least 1 of them)

yes

### Description of What Your Contract Does

Minimal ERC20

### Additional Details

On latest zksync foundry (`forge 0.0.2 (5d4bdb6 2024-11-07T00:33:22.708404000Z)`)

Running `forge create --zksync src/Counter.sol:Counter --rpc-url https://api.testnet.abs.xyz --private-key $PRIVATE_KEY` returns the following output:

```
[⠃] Using zksolc-1.5.4
[⠊] Compiling (zksync)
No files changed, compilation skipped
Error: contract was not deployed
```

The contract is in fact deployed.

<img width="1306" alt="image" src="https://github.com/user-attachments/assets/d7f75a57-7a6b-4257-8d99-f82e5f0948cd">
<img width="670" alt="image" src="https://github.com/user-attachments/assets/735de57e-3dc1-4f94-a6f6-687f87a5846f">

**Comment by alhenaai** [comment]
It appears that you are encountering an issue where the `forge create` command reports that the contract was not deployed, even though it actually was on the ZKsync Testnet. This could be due to a discrepancy in the feedback mechanism of the tool or a delay in the state update that the tool checks against.

Here are a few steps you can take to troubleshoot and potentially resolve this issue:

1. **Verify Deployment**: Double-check the deployment status on the ZKsync Testnet block explorer to confirm that the contract is indeed deployed. This will help isolate the issue to the tool's reporting mechanism.

2. **Check for Updates**: Ensure that your `foundry-zksync` toolchain is up to date. Sometimes, bugs related to deployment reporting are fixed in newer versions.

3. **Recompile Contract**: Force a recompilation of your contract by making a trivial change in the code or by deleting the cache. This might help if the issue is related to the compilation step being skipped.

4. **Review Tool Configuration**: Check your `foundry.toml` configuration file for any settings that might affect deployment behavior, especially those related to network settings and compiler versions.

5. **Consult Documentation and Community**: Review the [foundry-zksync documentation](https://docs.zksync.io/build/tooling/foundry/getting-started) for any known issues or peculiarities with deployment feedback. You can also seek advice by discussing this issue in the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

If the problem persists, it might be beneficial to report this as a bug to the foundry-zksync maintainers, providing them with all the details you've gathered. This could help improve the tool and assist others facing the same issue.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment]
@coffeexcoin was able to reproduce - investigating!

**Reply by dutterbutter** [reply]
Odd that it works fine for ZKsync Era RPC but not Abstract (e.g. `https://api.testnet.abs.xyz`)

**Reply by cygaar** [reply]
Could it be a node version thing?

**Reply by dutterbutter** [reply]
No, I dont think so as I am able to deploy / get receipt using HH fine. Seems specific to foundry-zksync I believe.

**Reply by dutterbutter** [reply]
Yes, seems the latest upstream changes that we merged is the issue. Still zeroing in but for instance compiling `forge` from commit `d9babce50528057735a52abb29f410e7c06e765f` and deploying  / receipt response works fine. Will continue investigating / work on fix. Thanks for posting 🙏

**Reply by dutterbutter** [reply]
Quick update: I've identified the immediate issue, which stems from an error in `get_receipt`: `TransportError(DeserError { err: Error("duplicate field status", line: 1, column: 3948)`. I'm still investigating the root cause to pinpoint where this duplication originates. I'll provide more details or confirm the fix as soon as I have an update.

**Reply by nbaztec** [reply]
Hi @coffeexcoin this is related to a recent alloy update that bars both the `root` and `status` field to be present on the transaction receipt as per [EIP-658](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-658.md). This was already included in zksync-era via https://github.com/matter-labs/zksync-era/pull/3187 and subsequently in era test node. Please update your rpc accordingly and `forge create` should work as usual.

**Reply by coffee-the-dev** [reply]
@nbaztec matter labs is currently responsible for this RPC.

**Reply by cygaar** [reply]
yeah, matter labs runs our infra - which team is in charge of updating our RPC?

**Reply by popzxc** [reply]
The team in charge of updating RPC has been notified and they will handle it.

**Reply by popzxc** [reply, marked as answer]
AFAIU the RPC was updated, so closing this discussion.
